src/algorithms/WState.cpp

```diff
@@ -0,0 +1,26 @@
+#include "algorithms/WState.hpp"
+
+namespace qc {
+void fGate(QuantumComputation& qc, std::size_t i, std::size_t j, std::size_t n,
+           std::size_t k) {
+  auto theta = std::acos(std::sqrt(1.0 / (n - k + 1)));
+  qc.ry(static_cast<Qubit>(j), -theta);
+  qc.z(static_cast<Qubit>(j), qc::Control{static_cast<Qubit>(i)});
+  qc.ry(static_cast<Qubit>(j), theta);
```

Comment:
I think the static casts here are no longer necessary since the latest changes. Might be wrong though. Could you remove them and see if clang-tidy complains?